src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -809,41 +807,69 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
             // SPEC: better than all other function members, then the function member invocation is ambiguous and a binding-time 
             // SPEC: error occurs.
 
-            // UNDONE: This is a naive quadratic algorithm; there is a linear algorithm that works. Consider using it.
+            int currentBestIndex;
             var candidates = result.Results;
-            for (int i = 1; i < candidates.Count; ++i)
+            for (currentBestIndex = 0; currentBestIndex < candidates.Count; currentBestIndex++)
+            {
+                if (candidates[currentBestIndex].Kind == OperatorAnalysisResultKind.Applicable)
+                {
+                    break;
+                }
+            }
+
+            if (currentBestIndex == candidates.Count)
+            {
+                return;
```

Comment:
> So it's safe to remove this check, right?

It depends on the input that we are getting. The ```if (result.GetValidCount() == 1)``` doesn't handle the case of no candidates marked as Applicable. If that set can contain candidates not marked as Inapplicable, then it looks like we might get into index out of range exception in the last loop in this function if we remove this early return.